packages/build/src/plugins_core/pre_cleanup/index.ts

```diff
@@ -1,28 +1,27 @@
 import { rm } from 'node:fs/promises'
+import { resolve } from 'node:path'
 
-import { scanForBlobs, getBlobsDirs } from '../../utils/blobs.js'
-import { CoreStep, CoreStepCondition, CoreStepFunction } from '../types.js'
+import { getBlobsDirs } from '../../utils/blobs.js'
+import { FRAMEWORKS_API_ENDPOINT } from '../../utils/frameworks_api.js'
+import { CoreStep, CoreStepFunction } from '../types.js'
 
 const coreStep: CoreStepFunction = async ({ buildDir, packagePath }) => {
-  const blobsDirs = getBlobsDirs(buildDir, packagePath)
+  const dirs = [...getBlobsDirs(buildDir, packagePath), resolve(buildDir, packagePath || '', FRAMEWORKS_API_ENDPOINT)]
+
   try {
-    await Promise.all(blobsDirs.map((dir) => rm(dir, { recursive: true, force: true })))
+    await Promise.all(dirs.map((dir) => rm(dir, { recursive: true, force: true })))
   } catch {
     // Ignore errors if it fails, we can continue anyway.
   }
 
   return {}
 }
 
-const blobsPresent: CoreStepCondition = async ({ buildDir, packagePath }) =>
-  Boolean(await scanForBlobs(buildDir, packagePath))
-
 export const preCleanup: CoreStep = {
   event: 'onPreBuild',
   coreStep,
   coreStepId: 'pre_cleanup',
   coreStepName: 'Pre cleanup',
   coreStepDescription: () => 'Cleaning up leftover files from previous builds',
-  condition: blobsPresent,
```

Comment:
This means that this core step will not run for all builds as opposed to only running if there are blobs directories. That is fine, since calling `rm()` on a directory that doesn't exist will not throw here, and we have a `try/catch` around the calls anyway.